starship

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import argparse
+import pdb
+
+from strawberryfields.engine import StarshipEngine
+from strawberryfields.io import load
+from strawberryfields import configuration
+
+PROMPTS = {
+    "hostname": "Please enter the hostname of the server to connect to: [{}] ",
+    "authentication_token": "Please enter the authentication token to use when connecting: [{}] ",
+    "save": "Would you like to save these settings to a local cofiguration file in the current "
+    "directory? [Y/n] ",
+}
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="run Strawberry Fields code on StarshipEngine")
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--input", "-i", help="the XBB file to run")
+    parser.add_argument(
+        "--output",
+        "-o",
+        help="where to output the result of the program - outputs to stdout by default",
+    )
+    parser.add_argument(
+        "--debug", action="store_true", help="returns a pdb shell after executing the program"
+    )
+    group.add_argument(
+        "--reconfigure",
+        action="store_true",
+        help="an interactive tool to reconfigure the API connection before executing the program",
+    )
+
+    args = parser.parse_args()
+
+    if args.reconfigure:
+        config = configuration.Configuration()
+
+        hostname = (
+            input(PROMPTS["hostname"].format(config.api["hostname"])) or config.api["hostname"]
+        )
+        authentication_token = (
+            input(PROMPTS["authentication_token"].format(config.api["authentication_token"]))
+            or config.api["hostname"]
+        )
+        save = input(PROMPTS["save"]).upper() == "Y"
+
+        if not save:
+            sys.stdout.write("Not writing configuration to file...")
+        else:
+            if not os.path.isfile("config.toml"):
+                sys.stdout.write("Writing configuration file to current working directory...\n")
+            else:
+                sys.stdout.write("Updating configuration in current working directory...\n")
+
+            config.api["hostname"] = hostname
+            config.api["authentication_token"] = authentication_token
+            config.save("config.toml")
+        sys.exit()
+
+    eng = StarshipEngine()
+    program = load(args.input)
+    eng.run(program)
+
+    if args.output_path:
+        with open(args.output_path, "w") as file:
+            file.write(str(eng.samples))
+            file.close()
+    else:
+        sys.stdout.write(str(eng.samples))
```

Comment:
Just curious the reason why you have decided to access `eng.samples` directly, as opposed to the
```python
results = eng.run(program)
print(results.samples)
```

Mostly because I've realised that `eng.samples` is not supposed to be user-facing, and should probably be a 'private' attribute.

Reply:
Thanks for noting this --- I will change. I think I saw someone using it (or perhaps it was in an example somewhere) and defaulted to that.

Reply:
Strange - do you remember where you saw it?

Reply:
I can't recall -- but it's possible that I am not remembering correctly. In any case, this is now fixed!